scripts/eosio_build.sh

```diff
@@ -30,375 +32,200 @@
-      mkdir -p $REPO_ROOT/tmp
-      TEMP_DIR="${REPO_ROOT}/tmp"
-      rm -rf $REPO_ROOT/tmp/*
-else # noexec wasn't found
-      TEMP_DIR="/tmp"
-fi
-
-export TMP_LOCATION=$TEMP_DIR
-
-function usage()
-{
-    cat >&2 <<EOT
-Usage: $0 OPTION...
+function usage() {
+   printf "Usage: $0 OPTION...
+  -P          Build with pinned clang and libcxx
   -o TYPE     Build <Debug|Release|RelWithDebInfo|MinSizeRel> (default: Release)
-  -p DIR      Prefix directory for dependencies & EOS install (default: $HOME)
+  -s NAME     Core Symbol Name <1-7 characters> (default: SYS)
   -b DIR      Use pre-built boost in DIR
+  -i DIR      Directory to use for installing dependencies & EOSIO (default: $HOME)
+  -y          Noninteractive mode (answers yes to every prompt)
   -c          Enable Code Coverage
   -d          Generate Doxygen
-  -s NAME     Core Symbol Name <1-7 characters> (default: SYS)
-  -y          Noninteractive mode (this script)
-  -P          Build with pinned clang 8 and libcxx
-  -f          Force rebuilding of dependencies
-  -m          Build MongoDB dependencies
-EOT
+   \\n" "$0" 1>&2
    exit 1
 }
 
+TIME_BEGIN=$( date -u +%s )
 if [ $# -ne 0 ]; then
-   while getopts ":cdo:s:p:b:mfPhy" opt; do
+   while getopts "o:s:b:i:ycdhmP" opt; do
       case "${opt}" in
          o )
             options=( "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
             if [[ "${options[*]}" =~ "${OPTARG}" ]]; then
-               CMAKE_BUILD_TYPE="${OPTARG}"
+               CMAKE_BUILD_TYPE=$OPTARG
             else
-               printf "\\nInvalid argument: %s\\n" "${OPTARG}" 1>&2
+               echo  "Invalid argument: ${OPTARG}" 1>&2
                usage
-               exit 1
             fi
          ;;
-         c )
-            ENABLE_COVERAGE_TESTING=true
-         ;;
-         d )
-            DOXYGEN=true
-         ;;
-         s)
+         s )
             if [ "${#OPTARG}" -gt 7 ] || [ -z "${#OPTARG}" ]; then
-               printf "\\nInvalid argument: %s\\n" "${OPTARG}" 1>&2
+               echo "Invalid argument: ${OPTARG}" 1>&2
                usage
-               exit 1
             else
-               CORE_SYMBOL_NAME="${OPTARG}"
+               CORE_SYMBOL_NAME=$OPTARG
             fi
-            ;;
-         b)
-             BOOST_ARG=$OPTARG
-             ;;
-         p)
-             PREFIX=$OPTARG
-             ;;
-         P)
-            PIN_COMPILER=true
-            ;;
-         h)
-            usage
-            exit 1
          ;;
-         y)
-            NONINTERACTIVE=1
+         b )
+            BOOST_LOCATION=$OPTARG
+         ;;
+         i )
+            INSTALL_LOCATION=$OPTARG
+         ;;
+         y )
+            NONINTERACTIVE=true
+            PROCEED=true
+         ;;
+         c )
+            ENABLE_COVERAGE_TESTING=true
+         ;;
+         d )
+            ENABLE_DOXYGEN=true
+         ;;
+         m )
+            ENABLE_MONGO=true
          ;;
-         f)
```

Comment:
We should probably add a `-f` back here that does nothing (but only for the release 1.8.x version not the develop one). Because existing users of 1.8.0 release candidates may expect `-f` to be accepted.